common/fixation_entry_index.go

```diff
@@ -26,11 +30,14 @@ func (fs *FixationStore) getEntryIndexStore(ctx sdk.Context) *prefix.Store {
 }
 
 // setEntryIndex stores an Entry index in the store
-func (fs FixationStore) setEntryIndex(ctx sdk.Context, safeIndex string) {
+func (fs FixationStore) setEntryIndex(ctx sdk.Context, safeIndex string, live bool) {
 	types.AssertSanitizedIndex(safeIndex, fs.prefix)
 	store := fs.getEntryIndexStore(ctx)
-	appendedValue := []byte(safeIndex) // convert the index value to a byte array
-	store.Set(types.KeyPrefix(safeIndex), appendedValue)
+	value := types.EntryIndexLive
+	if !live {
+		value = types.EntryIndexDead
```

Comment:
why do we need to save dead entry? i dont see any use for it, we always fetch the live ones

Reply:
it is a marker that there are still referenced (or about to be stale) entries of this index.
it's useful e.g. for migrations, to find all the entry versions (regardless of state) in the store.
this is also useful for testing and debugging.